src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -161,30 +164,37 @@ public class ConsensusManager : IDisposable
         /// A list of headers are presented from a given peer,
         /// we'll attempt to connect the headers to the tree and if new headers are found they will be queued for download.
         /// </summary>
-        /// <param name="networkPeerId">The peer that is providing the headers.</param>
+        /// <param name="peer">The peer that providing the headers.</param>
         /// <param name="headers">The list of new headers.</param>
-        /// <returns>The last chained header that is connected to the tree.</returns>
+        /// <param name="triggerDownload">Specifies if the download should be scheduled for interesting blocks.</param>
```

Comment:
Interesting blocks will be determined by the tree no?

Reply:
Yes

Reply:
This is the complexity we had to get because of BP?

Reply:
Yes